sale_order_price_recalculation/models/sale_order.py

```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+# (c) 2014 Carlos Sánchez Cifuentes <csanchez@grupovermon.com>
+# (c) 2015 Pedro M. Baeza <pedro.baeza@serviciosbaeza.com>
+# (c) 2015 Oihane Crucelaegui <oihanecrucelaegi@avanzosc.es>
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from openerp import models, api
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    @api.multi
+    def recalculate_prices(self):
+        for line in self.mapped('order_line'):
+            order = line.order_id
+            res = line.product_id_change(
+                order.pricelist_id.id, line.product_id.id,
+                qty=line.product_uom_qty, uom=False,
+                qty_uos=line.product_uos_qty, uos=False, name=line.name,
+                partner_id=order.partner_id.id, lang=False, update_tax=True,
+                date_order=order.date_order, packaging=False,
+                fiscal_position=order.fiscal_position.id, flag=False)
```

Comment:
I've just tested this on the runbot and I confirm it does not work

Steps to reproduce:

0. activate "manage UoM" in sales settings
1. create a sale order
2. add a sale order line for 2 dozens TONER
3. click "recalculate prices"

Expected result:

* nothing changes

Actual result: 

* the sale order is updated to a sale of 2 Unites TONER


==> needs fixing.